lib/goodreads_parser.py

```diff
@@ -289,9 +290,8 @@ def fullname_to_possible_aliases(self, fullname):
                     aliases.append("%s %s" % (title, surname))
             # Don't want the formats {ChristianName}, {Surname} and {ChristianName} {Lastname} in special cases
             # i.e. The Lord Ruler should never have "The Ruler", "Lord" or "Ruler" as aliases
-            if christian_name in self.COMMON_WORDS:
-                aliases.append('{0} {1}'.format(' '.join(parts), surname))
-            else:
+            # Same for John the Great
+            if christian_name not in self.COMMON_WORDS and (len(parts) == 0 or parts[0] not in self.COMMON_WORDS):
```

Comment:
General comment: this function seems very likely to get broken as we add more special cases, but also seems ripe for regression testing.  ie. we can throw a list of known names at it, and check it produces the right set of aliases.  Not something for now, obviously.